### PR TITLE
add simple string splitter processor

### PR DESCRIPTION
**Pull request by ekneg54**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,7 @@
 * Provide the possibility to consume lists, rules and configuration from files and http endpoints
 * Add `requester` processor that enriches by making http requests with field values
 * Add `calculator` processor to calculate with or without field values
+* Add `string_splitter` processor to split strings with variant length to lists
```
Comment (urdit52) [inline CHANGELOG.md:19]:
```suggestion
* Add `string_splitter` processor to split strings of variable length into lists
```

File: doc/source/development/notebooks/processor_examples/string_splitter.ipynb [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### The challenges\n",
+    "\n",
+    "- I want to split strings in a source field with variant field length\n"
```
Comment (urdit52) [inline doc/source/development/notebooks/processor_examples/string_splitter.ipynb:20]:
```suggestion
    "- I want to split strings of varying length contained in a source field\n"
```

File: doc/source/development/notebooks/processor_examples/string_splitter.ipynb [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from copy import deepcopy\n",
+    "\n",
+    "mydocument = deepcopy(document)\n",
+    "processor.process(mydocument)\n"
```
Comment (urdit52) [inline doc/source/development/notebooks/processor_examples/string_splitter.ipynb:191]: I find this example to be confusing. For me, it would be a lot clearer if the source and target field weren't identical.

File: logprep/processor/string_splitter/processor.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+"""
+
+from logprep.processor.field_manager.processor import FieldManager
+from logprep.processor.string_splitter.rule import StringSplitterRule
+from logprep.util.helper import add_field_to, get_dotted_field_value
+from logprep.processor.base.exceptions import DuplicationError, ProcessingWarning
+
+
+class StringSplitter(FieldManager):
+    """A processor that ..."""
```
Comment (urdit52) [inline logprep/processor/string_splitter/processor.py:29]: there seems to be something missing here.

File: logprep/processor/string_splitter/processor.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    def _apply_rules(self, event: dict, rule: StringSplitterRule):
+        target_field = rule.target_field
+        source_field = rule.source_fields[0]
+        source_field_content = get_dotted_field_value(event, source_field)
+        if not isinstance(source_field_content, str):
+            raise ProcessingWarning(
+                f"{ self.describe() }: source_field '{source_field}' is not a string"
+            )
+        result = source_field_content.split(rule.delimeter)
+        successfull = add_field_to(
```
Comment (urdit52) [inline logprep/processor/string_splitter/processor.py:42]:
```suggestion
        successful = add_field_to(
```

File: logprep/processor/string_splitter/processor.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+            event=event,
+            output_field=target_field,
+            content=result,
+            extends_lists=rule.extend_target_list,
+            overwrite_output_field=rule.overwrite_target,
+        )
+        if not successfull:
```
Comment (urdit52) [inline logprep/processor/string_splitter/processor.py:49]:
```suggestion
        if not successful:
```

for correct spelling.

File: tests/unit/processor/string_splitter/test_string_splitter.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    (
+        "splits without delimeter on whitespace",
+        {
+            "filter": "message",
+            "string_splitter": {"source_fields": ["message"], "target_field": "result"},
+        },
+        {"message": "this is the message"},
+        {"message": "this is the message", "result": ["this", "is", "the", "message"]},
+    )
+]  # testcase, rule, event, expected
```
Comment (urdit52) [inline tests/unit/processor/string_splitter/test_string_splitter.py:17]: I think that we should also add a test case that uses any other delimiter that is not a whitespace.

**Review by urdit52 — CHANGES_REQUESTED** [review]
A few remarks. Could you please also explicitly test cases that use a defined delimiter?

**Review by urdit52 — APPROVED** [review]
thanks a lot.
